### PR TITLE
Cream & Yoghurt additions/reworks

### PR DESCRIPTION
**Pull request by CrystalClover**
[CATERING] [CHEMISTRY] [REWORK] [QoL]
### Additions:
Cream reagent: Taken from cream (condiment), breaks down into milk and cholesterol
Yoghurt reagent: Made from combining Milk and ????, heals damage to the intestine

### Reworks: 
Food Processor:
Now converts every 10 units of milk or yoghurt in a drinks container into cream and yoghurt (food) respectively, and dispenses the remaining container with any milk/yoghurt that wasn't used.

Frozen Yoghurt & Yoghurt: 
Both contain 10 units of Yoghurt, and give you a Disease Resist buff, Frozen Yoghurt also gives you the Cold Food buff

Condiments:
You cannot apply a condiment to a food if the condiment contains no reagents 
(does not affect things such as adding maple syrup to pancakes)
You cannot add a condiment to another condiment

Cream of Mushroom:
Now requires Cream instead of Milk, is cooked for 8 seconds on low in an oven (or 4 seconds on high)

The following foods have had their reagents changed:
Icecream - Starts with 10 units of Cream
Cream Pie - Starts with 10 units of Cream
Cream of Mushroom - Starts with 10 units of Cream
Creamy Sponge Cake - Starts with 20 units of Sugar & 10 units of Cream
All Cookies - Now contain chems that come from the constituant parts
All Moon Pies - Inherit the chems from the cookies used and also start with 10 units of Cream
Rancid Milk - Now contains 25 units of yoghurt and 25 units of ????

## Why's this needed? 
Makes cream and yoghurt much easier to obtain and gives them more uses, the items with changed reagents now make more sense than having none at all

## Changelog 

```changelog
(u)CrystalClover
(*)Added Cream reagent to the game, which comes from dollops of cream. Can be used for filling donuts or icing cakes!
(*)Added Yoghurt reagent to the game, which is made from mixing Milk and ????. Improves gut health! (Heals intestinal damage)
(*)Food processors no longer destroy the bottle they use to make Cream or Yoghurt! They also accept any drink container, and convert every 10 units of Milk/Yoghurt into dollops of Cream or Yoghurt pots respectively.
(+)Almost all foods made with Cream now have 10 units of the Cream reagent in them by default. (This also includes icecreams)
(+)Cookies  and Moon Pies have had their reagents changed to better reflect what they are made of.
(+)Cream of Mushroom Soup is now made with 1 Mushroom and 1 Cream, and is cooked in an over for 8 seconds on Low.
(+)Condiments can no longer be applied to eachother, and cannot be applied to a food if they do not contain anything.
```

### CODE REVIEW

File: code/modules/food_and_drink/ice_cream.dm [hunk 1]
```diff
@@ -100,12 +101,16 @@
 	icon = 'icons/obj/foodNdrink/food_snacks.dmi'
 	icon_state = "yoghurt"
 	required_utensil = REQUIRED_UTENSIL_SPOON
-	bites_left = 6
+	bites_left = 4
 	heal_amt = 1
+	initial_volume = 10
+	initial_reagents = list("yoghurt"=10)
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/ice_cream.dm:107]:
```suggestion
	initial_volume = 30
	initial_reagents = list("yoghurt"=15)
```
Food with spare reagent space, beloved by traitors and chefs alike!

File: code/modules/food_and_drink/snacks.dm [hunk 2]
```diff
@@ -348,14 +350,16 @@
 		heal_amt = 0
 		icon_state = "cookie-metal"
 		food_effects = list("food_hp_up")
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:353]:
```suggestion
		initial_volume = 40
```

File: code/modules/food_and_drink/snacks.dm [hunk 3]
```diff
@@ -348,14 +350,16 @@
...
+		initial_reagents = list("sugar" = 10, "iron" = 10)
 
 	chocolate_chip
 		name = "chocolate-chip cookie"
 		desc = "Invented during the Great Depression, this chocolate-laced cookie was a key element of FDR's New Deal policies."
 		icon_state = "cookie-chips"
 		heal_amt = 2
-		initial_volume = 15
-		initial_reagents = list("chocolate"=10)
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:361]:
```suggestion
		initial_volume = 40
```

File: code/modules/food_and_drink/snacks.dm [hunk 4]
```diff
@@ -367,39 +371,39 @@
 		name = "bacon cookie"
 		desc = "A cookie made out of bacon. Is this intended to be savory or a sweet candied bacon sort of thing? Whatever it is, it's pretty dumb."
 		icon_state = "cookie-bacon"
-		initial_volume = 50
-		initial_reagents = list("porktonium"=25)
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:374]:
```suggestion
		initial_volume = 50
```

File: code/modules/food_and_drink/snacks.dm [hunk 5]
```diff
@@ -367,39 +371,39 @@
...
+		initial_reagents = list("sugar" = 10, "porktonium"=10)
 		food_effects = list("food_sweaty")
 
 	jaffa
 		name = "jaffa cake"
 		desc = "Legally a cake, this edible consists of precision layers of chocolate, sponge cake, and orange jelly."
 		icon_state = "cookie-jaffa"
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:382]:
```suggestion
		initial_volume = 40
```

File: code/modules/food_and_drink/snacks.dm [hunk 6]
```diff
@@ -367,39 +371,39 @@
...
+		initial_reagents = list("sugar" = 10, "chocolate"=5, "juice_orange"=5)
 
 	spooky
 		name = "spookie"
 		desc = "Two ounces of pure terror."
 		icon_state = "cookie-spooky"
 		frosted = 1
-		initial_volume = 25
-		initial_reagents = list("ectoplasm"=10)
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:390]:
```suggestion
		initial_volume = 40
```

File: code/modules/food_and_drink/snacks.dm [hunk 7]
```diff
@@ -367,39 +371,39 @@
...
 		meal_time_flags = MEAL_TIME_FORBIDDEN_TREAT
 
 	butter
 		name = "butter cookie"
 		desc = "Little bite-sized heart attacks." //no kidding
 		icon_state = "cookie-butter"
 		frosted = 1
-		initial_volume = 25
-		initial_reagents = list("butter"=10)
+		initial_volume = 20
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:399]:
```suggestion
		initial_volume = 40
```

File: code/modules/food_and_drink/snacks.dm [hunk 8]
```diff
@@ -437,6 +441,8 @@
 	icon_state = "moonpie-sugar"
 	bites_left = 1
 	heal_amt = 6
+	initial_volume = 40
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:444]:
```suggestion
	initial_volume = 75
```

File: code/modules/food_and_drink/snacks.dm [hunk 9]
```diff
@@ -465,13 +471,17 @@
 		icon_state = "moonpie-metal"
 		heal_amt = 0
 		food_effects = list("food_hp_up_big")
+		initial_volume = 50
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:474]:
```suggestion
		initial_volume = 100
```

File: code/modules/food_and_drink/snacks.dm [hunk 10]
```diff
@@ -465,13 +471,17 @@
...
+		initial_reagents = list("sugar" = 20, "iron" = 20, "cream" = 10)
 
 	chocolate_chip
 		name = "chocolate-chip moon pie"
 		desc = "The confection commonly credited with winning the Korean, Gulf, and Unfolder wars."
 		icon_state = "moonpie-chips"
 		heal_amt = 7
 		food_effects = list("food_refreshed_big")
+		initial_volume = 50
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:483]:
```suggestion
		initial_volume = 100
```

File: code/modules/food_and_drink/snacks.dm [hunk 11]
```diff
@@ -486,21 +496,25 @@
 		icon_state = "moonpie-bacon"
 		heal_amt = 5
 		initial_volume = 50
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:498]:
```suggestion
		initial_volume = 100
```

File: code/modules/food_and_drink/snacks.dm [hunk 12]
```diff
@@ -486,21 +496,25 @@
...
+		initial_volume = 50
+		initial_reagents = list("sugar" = 20, "chocolate" = 10, "juice_orange" = 10, "cream" = 10)
 		food_effects = list("food_refreshed_big")
 
 	chocolate
 		name = "whoopie pie"
 		desc = "A confection infamous for being especially terrible for you, in a culture noted for having nothing but foods that are terrible for you."
 		icon_state = "moonpie-chocolate"
 		heal_amt = 25 //oh jesus
+		initial_volume = 60
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:516]:
```suggestion
		initial_volume = 100
```

File: code/modules/food_and_drink/snacks.dm [hunk 13]
```diff
@@ -486,21 +496,25 @@
...
-		initial_reagents = "porktonium"
+		initial_reagents = list("sugar" = 20, "porktonium" = 20, "cream" = 10)
 		food_effects = list("food_sweaty_big")
 
 	jaffa
 		name = "jaffa moon cobbler"
 		desc = "This dish was named in an attempt to dodge sales taxes on pie production. However, it is actually legally considered a form of crumble."
 		icon_state = "moonpie-jaffa"
 		heal_amt = 8
+		initial_volume = 50
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:507]:
```suggestion
		initial_volume = 100
```

File: code/modules/food_and_drink/snacks.dm [hunk 14]
```diff
@@ -509,8 +523,8 @@
 		icon_state = "moonpie-spooky"
 		heal_amt = 6
 		frosted = 1
-		initial_volume = 25
-		initial_reagents = list("ectoplasm"=10)
+		initial_volume = 50
```
Comment (MylieDaniels) [inline code/modules/food_and_drink/snacks.dm:526]:
```suggestion
		initial_volume = 100
```

**Review by MylieDaniels — CHANGES_REQUESTED** [review]
Looks really good, though a bit broad!  I've made suggestions, mostly about keeping some empty space in the food proportional to it's reagent volume - this helps people make more customized food as well as allowing poisonings to occur.

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 15]
```diff
@@ -781,6 +781,15 @@
 				boutput(M, "<span class='alert'>A horrible smell assaults your nose! What in space is it?</span>")
 			return
 
+	yoghurt
+		name = "Yoghurt"
+		id = "yoghurt"
+		result = "yoghurt"
+		required_reagents = list("milk" = 1, "yuck" = 1)
+		result_amount = 1
```
[the remark this answers is not in the document]
Comment (CrystalClover) [reply code/modules/chemistry/Chemistry-Recipes.dm:789]: changed resulting amount to 2

**Review by MylieDaniels — APPROVED** [review]
Looks solid to me!

**Review by kbsmilk — APPROVED** [review]
Looks fine code-wise. In the future I'd suggest atomizing some more. The cream obj and its addition to existing foods, changing a bunch of cookie reagents, and making condiments unusable on other condiments: all three of these could be separated by scope, into 3 different PRs.

**Comment by kbsmilk** [comment]
I forgot to mention, the changelog is way too big IMO. At most, I'd suggest 1 major and 1 minor, or just 2 minor, explaining cream and yoghurt's additions and a brief line about their recipe.

**Pull request merged**